src/test/scala/org/broadinstitute/dsde/workbench/sam/api/TestSamRoutes.scala

```diff
@@ -68,11 +74,38 @@ object TestSamRoutes {
   )
 
   def apply(resourceTypes: Map[ResourceTypeName, ResourceType], userInfo: UserInfo = defaultUserInfo, policyAccessDAO: Option[AccessPolicyDAO] = None, policies: Option[mutable.Map[WorkbenchGroupIdentity, WorkbenchGroup]] = None)(implicit system: ActorSystem, materializer: Materializer, executionContext: ExecutionContext, contextShift: ContextShift[IO]) = {
+    val dbRef = TestSupport.dbRef
+    val directoryConfig: DirectoryConfig = TestSupport.directoryConfig
+    val dirURI = new URI(directoryConfig.directoryUrl)
+    val ldapConnectionPool = new LDAPConnectionPool(new LDAPConnection(dirURI.getHost, dirURI.getPort, directoryConfig.user, directoryConfig.password), directoryConfig.connectionPoolSize)
     val resourceTypesWithAdmin = resourceTypes + (resourceTypeAdmin.name -> resourceTypeAdmin)
     // need to make sure MockDirectoryDAO and MockAccessPolicyDAO share the same groups
     val groups: mutable.Map[WorkbenchGroupIdentity, WorkbenchGroup] = policies.getOrElse(new TrieMap())
-    val directoryDAO = new MockDirectoryDAO(groups)
-    val registrationDAO = new MockDirectoryDAO()
+    val directoryDAO = new MockDirectoryDAO(groups) {
```

Comment:
Do we need to override `checkStatus`? Getting rid of these overrides would let us remove the `LdapConnectionPool` that you're concerned about as well.

I tried to test it myself, but I'm having all kinds of environment issues

Reply:
Also, in general, I would love to avoid overriding `checkStatus` in multiple places if possible. I'd prefer to see us either use a real `LDAPRegistrationDAO` or to use the mocked out `checkStatus`. Duplicating the implementation will make it difficult to change the `checkStatus` implementation in the future

Reply:
So, I could just have the mock return true on status, which would satisfy the tests, I believe. I can't have the mock actually have an LdapConnectionPool because it doesn't have a real db. But I think that is fine.

Reply:
I'm fine with the mock DAOs defaulting to true with individual overrides to return false as needed/when appropriate